lib/api/apiUtils/authorization/prepareRequestContexts.js

```diff
@@ -25,8 +28,15 @@ export default function prepareRequestContexts(apiMethod, request, sourceBucket,
     if (apiMethod === 'multiObjectDelete' || apiMethod === 'bucketPut') {
         return null;
     }
+    if (apiMethodWithVersion[apiMethod] && request.query &&
+      request.query.versionId) {
+        apiMethodAfterVersionCheck = `${apiMethod}Version`;
+    } else {
+        apiMethodAfterVersionCheck = apiMethod;
+    }
     const requestContexts = [];
-    if (apiMethod === 'objectCopy' || apiMethod === 'objectPutCopyPart') {
+    if (apiMethodAfterVersionCheck === 'objectCopy' ||
+    apiMethodAfterVersionCheck === 'objectPutCopyPart') {
         // TODO: Categorize objectCopy requests with copy source version
         // ids as 'objectGetVersion' action (after related PR is merged)
```

Comment:
add a check if `sourceVersionId !== undefined`, `'objectGet'` below (line 47) should become `objectGetVersion` api method.